### ISSUE TITLE
Firefox: various web sites don't work

### ISSUE THREAD

**Issue by clamage45** [EFForg/privacybadger#1478]
My system: Windows 10 with latest updates. Firefox any recent version, currently v54.0.1.

With privacybadger enabled, some web sites don't work properly. Examples:
1. Yelp.com -- Go to a restaurant that allows online ordering, and click the start-order or menu button. I get a wait symbol that hangs forever. But if I refresh the page (circular arrow in address bar), the new page opens.
2. discoverbank.com -- Try to link an external account. Fill in the data, click continue, nothing happens. In this case, refreshing the page did not help.
3. My doctor's patient portal -- After logging in, click a button to select a service, nothing happens, refreshing the page does not help.
4. http://websudoku.com/ -- Firefox locks up for about 15 seconds after loading the page, then is OK.

After disabling privacybadger, these pages all work fine.

**Comment by ghostwords** [comment]
Sorry about the trouble! There might be several different bugs going on here ... Would you mind opening separate issues for each website? For each issue, please let us know if disabling Privacy Badger just for the site (by clicking that button in Badger's popup) and reloading the page fixes the problem or not.

**Comment by clamage45** [comment]
Let's just go with the yelp.com problem. Disabling PrivacyBager via its "disable on this site" button made the problem go away. Here is a specific web site to demonstrate the problem:
https://www.yelp.com/biz/mi-zacatecas-hawthorne
Click the big green button "view menu".  You should see a "wait" indicator that stays forever.
Refresh the page (circular arrow in the location bar), and the new page appears.d

**Comment by ghostwords** [comment]
Which domains does your Badger report as blocked on the Yelp page? Just the ones with the red toggle. Restaurant looks good by the way ...

**Comment by clamage45** [comment]
The PB icon shows the number 10. PB says 14 potential trackers. The list of 14 trackers shows 8 in red marked blocked:
6372968.fls.doubleclick.net
connect.facebook.net
www.google-analytics.com 
secure-us.imrworldwide.com
secure.quantserve.com
ei.rlcdn.com
idsync.rlcdn.com
sb.scorecardresearch.com

BTW, making this list is really tedious, because the list disappears with every click.

**Comment by ghostwords** [comment]
Ah I see what's happening on Yelp, this is a Firefox-specific issue that doesn't have anything to do with what's blocked. I think it's covered by this Bugzilla ticket: https://bugzilla.mozilla.org/show_bug.cgi?id=1373776. Thanks for the report!

**Comment by ghostwords** [comment]
To confirm: If you disable all other add-ons for now, and install my [test add-on](https://github.com/EFForg/privacybadger/files/1079120/firefox_webext_onunload_bug_demo.zip), do you still run into the issue?

To load the test add-on in Firefox, unzip the archive somewhere, visit about:debugging, click on Load Temporary Add-on, and pick the manifest.json file inside the unzipped folder.

**Comment by ghostwords** [comment]
This might be fixed now that Firefox 55 reached stable.

**Comment by ghostwords** [comment]
Might be a duplicate of #1438. Let me know if this is not the case.

**Comment by clamage45** [comment]
Thanks! I look forward to a Firefox update. :-)

Possibly related: Gmail used to work flawlessly in Firefox. Recently, it
sometimes gets stuck. Typical failure mode: A message is open, but nothing
happens when I click on anything in the entire window. (Can't close or
reply to or delete the message, can't switch to different folder, etc.) I
can switch to a different tab, so Firefox is not completely hung.  If I
close the gmail tab and re-open, things are fine again, but when closing
the tab, I get a pop-up "Are you sure you want to leave this page? Changes
you made could be lost." So apparently Firefox is waiting for something.
This problem  is intermittent, but happens at least once per day.

Do you think this is similar to the problem with Yelp? If not, I'll file an
issue with Firefox.

**Comment by ghostwords** [comment]
It could be ... Try downloading the latest stable Firefox (55) and testing there.

**Comment by ghostwords** [comment]
I'm going to resolve this. Please feel free to open new issues for any additional problems or questions.
